llvm/lib/FileCheck/FileCheck.cpp

```diff
@@ -2344,13 +2350,13 @@ size_t FileCheckString::CheckDag(const SourceMgr &SM, StringRef Buffer,
   // group, so we don't use a range-based for loop here.
   for (auto PatItr = DagNotStrings.begin(), PatEnd = DagNotStrings.end();
        PatItr != PatEnd; ++PatItr) {
-    const Pattern &Pat = *PatItr;
+    const Pattern &Pat = PatItr->DagNotPat;
     assert((Pat.getCheckTy() == Check::CheckDAG ||
             Pat.getCheckTy() == Check::CheckNot) &&
            "Invalid CHECK-DAG or CHECK-NOT!");
 
     if (Pat.getCheckTy() == Check::CheckNot) {
-      NotStrings.push_back(&Pat);
+      NotStrings.emplace_back(Pat, PatItr->DagNotPrefix);
```

Comment:
Why not make NotString a std::vector<DagNotPrefixInfo *> and just push_back(&*PatItr)? It avoids creating a new instance here as well.

Reply:
That really completely escaped my guard. I will keep that in mind, and fix this too.